### PR TITLE
Conect api

### PR DESCRIPTION
**Pull request by KaskMIL**
# In this new Pull Request I'm adding the following features:
- Refactor BooksAsync.js to use createSlice and createAsyncThunk.
- Create an async action creator to fetch the list from API.
- Refactor addBook action creator to add new books to the API.
- Refactor deleteBook action creator to delete a book from API.
- Refactor components to work with the data fetched from the API.

### CODE REVIEW

File: src/components/BookList/BookList.js [hunk 1]
```diff
@@ -1,15 +1,27 @@
-import React from 'react';
-
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+// import { baseUrl } from '../../redux/Books/BooksRedux';
```
Comment (t-yanick) [inline src/components/BookList/BookList.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/redux/Books/BooksRedux.js [hunk 2]
```diff
@@ -28,3 +35,20 @@ export const deleteBook = (id) => ({
...
+  fetch(`${baseUrl}`, {
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then((response) => response.json())
+    .then((json) => dispatch(updateState(json)));
+};
+ */
```
Comment (t-yanick) [inline src/redux/Books/BooksRedux.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @KaskMIL, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great job with  createAsyncThunk, createSlice.
:heavy_check_mark: Descriptive `PR` summary.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KaskMIL** [comment]
Hi @t-yanick 
Thanks for your review, I'll implement the changes that you suggest
Have a great day, cheers!

File: src/redux/intialData.js [hunk 3]
```diff
@@ -1,23 +1,5 @@
-import { v4 } from 'uuid';
-
 const initialData = {
```
Comment (rustam-tolipov) [inline src/redux/intialData.js:1]: - [ ] [OPTIONAL] You forgot to create the MIT license file. Kindly add it to your repo 👍🏼

**Review by rustam-tolipov — APPROVED** [review]
## Hello @KaskMIL  👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**